jmix-core/core/src/main/java/io/jmix/core/MetadataTools.java

```diff
@@ -708,20 +718,22 @@ public Collection<MetaClass> getAllJpaEmbeddableMetaClasses() {
     }
 
     /**
-     * @return collection of all Java enums used as a type of an entity attribute
+     * @return collection of all Java enums implementing {@link io.jmix.core.metamodel.datatype.EnumClass}
      */
     public Collection<Class<?>> getAllEnums() {
         if (enums == null) {
             synchronized (this) {
-                enums = new HashSet<>();
-                for (MetaClass metaClass : metadata.getSession().getClasses()) {
-                    for (MetaProperty metaProperty : metaClass.getProperties()) {
-                        if (metaProperty.getRange().isEnum()) {
-                            Class<?> c = metaProperty.getRange().asEnumeration().getJavaClass();
-                            enums.add(c);
-                        }
+                Set<Class<?>> tmpEnums = new HashSet<>();
```

Comment:
It is not directly related to the current fix, but since we're inside this method, it looks that the second null check statement is missing inside the synchronized block.

```java
        if (enums == null) {
            synchronized (this) {
                if (enums == null) {
```

Reply:
Makes sense. Done.